test/parallel/test-child-process-stdout-flush-exit.js

```diff
@@ -30,17 +29,11 @@ if (process.argv[2] === 'child') {
   // check if we receive both 'hello' at start and 'goodbye' at end
   child.stdout.setEncoding('utf8');
   child.stdout.on('data', function(data) {
-    if (data.slice(0, 6) == 'hello\n') {
-      gotHello = true;
-    } else if (data.slice(data.length - 8) == 'goodbye\n') {
-      gotBye = true;
-    } else {
-      gotBye = false;
-    }
+    stdout += data;
   });
 
-  child.on('close', function(data) {
-    assert(gotHello);
-    assert(gotBye);
+  child.on('close', function() {
+    assert.equal(stdout.slice(0, 6), 'hello\n');
+    assert.equal(stdout.slice(stdout.length - 8), 'goodbye\n');
   });
```

Comment:
Can you wrap the callback in a `common.mustCall(...)` while you're here?  LGTM otherwise.